### PR TITLE
[SPARK-7532][STREAMING] StreamingContext.start() made to logWarning and not throw exception

### PR DESCRIPTION
**Pull request by tdas**

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -514,12 +514,10 @@ class StreamingContext private[streaming] (
 
   /**
    * Start the execution of the streams.
-   *
-   * @throws SparkException if the context has already been started or stopped.
    */
   def start(): Unit = synchronized {
     if (state == Started) {
-      throw new SparkException("StreamingContext has already been started")
+      logWarning("StreamingContext has already been started")
```
Comment (zsxwing) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Need to add "return" here.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Yep, missed that while splitting this out from a different PR.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: On second thought, I will fix this when #6058 goes in.

**Comment by tdas** [comment]
test this please.

**Comment by tdas** [comment]
@zsxwing Can you take a look at this.

**Comment by zsxwing** [comment]
LGTM

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 2]
```diff
@@ -527,29 +527,26 @@ class StreamingContext private[streaming] (
 
   /**
    * Start the execution of the streams.
-   *
-   * @throws SparkException if the context has already been started or stopped.
```
Comment (zsxwing) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: I think we still need to mention `SparkException` here.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Right. I need to modify it to say SparkException will be thrown only when the context has been already stopped.

**Comment by tdas** [comment]
I am merging this.

**Pull request closed without merging**